File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/SurveyAndQuestions.java
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import jakarta.validation.constraints.*;
import java.io.Serializable;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 * SurveyAndQuestions.
 */
@lombok.Getter
@lombok.Setter
public class SurveyAndQuestions implements Serializable {

    @NotNull
    private Long surveyId;

    @NotNull
    @Size(max = 254)
    private String name;

    @NotNull
    @Size(max = 254)
    private String title;

    @NotNull
    private Long hospitalId;

    @NotNull
    @Size(max = 3)
    private String lang;

    private List<SurveySectionOutput> sections = new ArrayList<>();

    public SurveyAndQuestions addSections(SurveySectionOutput sections) {
        this.sections.add(sections);
        return this;
    }
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/dtos/SurveyAndQuestions.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service.dtos

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import jakarta.validation.constraints.*
import java.io.Serializable
import java.math.*
import java.time.*

/** SurveyAndQuestions. */
data class SurveyAndQuestions(
    @NotNull val surveyId: Long? = null,
    @NotNull @Size(max = 254) val name: String? = null,
    @NotNull @Size(max = 254) val title: String? = null,
    @NotNull val hospitalId: Long? = null,
    @NotNull @Size(max = 3) val lang: String? = null,
    val sections: MutableList<SurveySectionOutput> = mutableListOf(),
) : Serializable {

    fun addSections(sections: SurveySectionOutput): SurveyAndQuestions {
        this.sections += sections
        return this
    }
}
